LeanOA/ForMathlib/Analysis/CStarAlgebra/SpecialFunctions/Abs.lean

```diff
@@ -0,0 +1,172 @@
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Instances
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Order
+
+/-!
+# Absolute value of an operator defined via the continuous functional calculus
+
+This file provides basic API for `CFC.abs` for `CStarAlgebra`s.
+
+-/
+
+open scoped NNReal
+
+namespace CFC
+
+section NonUnital
+
+variable {A : Type*} [NonUnitalNormedRing A] [StarRing A]
+  [Module ℝ A] [SMulCommClass ℝ A A] [IsScalarTower ℝ A A]
+  [NonUnitalContinuousFunctionalCalculus ℝ (IsSelfAdjoint : A → Prop)]
+
+lemma mul_self_eq_mul_self {a : A} (ha : IsSelfAdjoint a) : a * a =
+    cfcₙ (fun (x : ℝ) ↦ x * x) a := by
+  conv_lhs => rw [← cfcₙ_id' ℝ a, ← cfcₙ_mul ..]
+
+variable [UniqueNonUnitalContinuousFunctionalCalculus ℝ A]
+
+lemma cfcₙ_sqrt_mul_self {a : A} (ha : IsSelfAdjoint a) :
+    cfcₙ Real.sqrt (a * a) = cfcₙ (fun x ↦ √(x * x)) a := by
+  rw [mul_self_eq_mul_self ha, ← cfcₙ_comp a (f := fun x ↦ x * x) (g := fun x ↦ √x),
+     Function.comp_def]
+
+variable {A : Type*} [PartialOrder A] [NonUnitalNormedRing A] [StarRing A]
+   [Module ℝ A] [SMulCommClass ℝ A A] [IsScalarTower ℝ A A]
+   [NonUnitalContinuousFunctionalCalculus ℝ (IsSelfAdjoint : A → Prop)]
+   [UniqueNonUnitalContinuousFunctionalCalculus ℝ≥0 A]
+   [StarOrderedRing A] [NonnegSpectrumClass ℝ A]
+
+lemma sqrt_eq_cfcₙ_real_sqrt {a : A} (ha : 0 ≤ a := by cfc_tac) :
+    CFC.sqrt a = cfcₙ Real.sqrt a := by
+  rw [sqrt_eq_iff _ (hb := cfcₙ_nonneg (A := A) (fun x _ ↦ Real.sqrt_nonneg x)),
+    ← cfcₙ_mul ..]
+  conv_rhs => rw [← cfcₙ_id (R := ℝ) a]
+  refine cfcₙ_congr fun x hx ↦ ?_
+  refine Real.mul_self_sqrt ?_
+  exact quasispectrum_nonneg_of_nonneg a ha x hx
+
+variable {A : Type*} [NonUnitalCStarAlgebra A]
+
+lemma cfcₙ_sq {a : A} {f : ℂ → ℂ} (hf : ContinuousOn f (quasispectrum ℂ a) := by cfc_cont_tac) (hf0 : f 0 = 0 := by cfc_zero_tac):
+  cfcₙ (fun z : ℂ ↦ ((f z) ^ 2 : ℂ)) a = (cfcₙ f a) * (cfcₙ f a) := by
+  rw [← cfcₙ_mul ..]
+  simp only [Complex.norm_eq_abs, sq]
+
+lemma cfcₙ_sq' {a : A} {f : ℂ → ℂ} (hf : ContinuousOn f (quasispectrum ℂ a) := by cfc_cont_tac) (hf0 : f 0 = 0 := by cfc_zero_tac):
+  cfcₙ (fun z : ℂ ↦ ((f z) ^ 2 : ℂ)) a = cfcₙ (fun z : ℂ ↦ ((f z) * (f z) : ℂ)) a := by
+  rw [cfcₙ_sq ..]
+  apply Eq.symm (cfcₙ_mul f f a hf hf0 hf hf0)
+
+variable [PartialOrder A] [StarOrderedRing A]
+
+lemma abs_eq_cfcₙ_norm {a : A} (ha : IsSelfAdjoint a) :
+    abs a = cfcₙ (‖·‖) a := by
+   simp only [abs, Real.norm_eq_abs, ← Real.sqrt_sq_eq_abs, sq]
+   rw [sqrt_eq_cfcₙ_real_sqrt (star_mul_self_nonneg a), ha.star_eq, cfcₙ_sqrt_mul_self ha]
+
+lemma abs_eq_zero_iff {a : A} : abs a = 0 ↔ a = 0 := by
+  rw [abs, sqrt_eq_zero_iff _, CStarRing.star_mul_self_eq_zero_iff]
+
+@[aesop safe apply (rule_sets := [CStarAlgebra])]
+theorem IsSelfAdjoint.mul_self_nonneg {a : A} (ha : IsSelfAdjoint a) : 0 ≤ a * a := by
+  simpa [ha.star_eq] using star_mul_self_nonneg a
```

Comment:
This is in Mathlib already (maybe it isn't before the bump, I'm not sure), but right now you wouldn't know because the name is wrong. Currently, it's `CFC.IsSelfAdjoint.mul_self_nonneg` instead of `_root_.IsSelfAdjoint.mul_self_nonneg`. You should just be able to delete this and everything will compile.